### PR TITLE
Update Makeflow's JX interface

### LINKED ISSUE #1776: JX command line args
Add option to pass JX arguments to Makeflow JX in conjunction with the existing arg file.

### PR DESCRIPTION
**Pull request by trshaffer**
Per discussion with @nhazekam, I renamed Makeflow's `--jx-context` option to `--jx-args`. I kept the old option for compatibility. I also added the `--jx-define` flag to add JX variables from the command line. For example,

    $ makeflow workflow.jx --jx-define splits=10

Note that the thing on the right is a JX expression, so you can pass strings, functions, etc. Watch out for shell characters, though.

    $ makeflow workflow.jx --jx-define task=\"foo\"
    $ makeflow workflow.jx --jx-define "ids=range(10)"

Multiple occurrences of `--jx-args` and `--jx-define` are applied in the order they appear.

Resolves #1776

### CODE REVIEW

File: makeflow/src/makeflow.c [hunk 1]
```diff
@@ -1856,23 +1890,14 @@ int main(int argc, char *argv[])
...
-			if (!t) fatal("failed to parse context");
-			ctx = jx_eval(t, NULL);
-			jx_delete(t);
-		}
 		if (jx_input) {
-			struct jx *t = dag;
-			dag = jx_eval(t, ctx);
-			jx_delete(t);
-			jx_delete(ctx);
+			jx_tmp = jx_eval(dag, jx_args);
```
[the remark this answers is not in the document]
Comment (trshaffer) [reply makeflow/src/makeflow.c]: It gets set to an empty object at the beginning, so shouldn't be an issue

**Comment by nhazekam** [comment]
Also, can you rename context to args in `jx2json` executable.

**Comment by dthain** [comment]
Once this is complete, please take a look at the makeflow-examples repository and update the examples and files (many named `context.jx`) to be consistent with the new nomenclature.

**Comment by nhazekam** [comment]
I have these changes applied and I am waiting for this to go through, along with the ceil and floor pull request. cooperative-computing-lab/makeflow-examples#16 This can be moved to multiple commits on makeflow-examples, but I think they are both in the same vein of work.

**Pull request merged**